### ISSUE TITLE
IOB error log flooded with Request Timeout error

### ISSUE THREAD

**Issue by MAR-L3Y** [Sickboy78/ioBroker.sureflap#43]
Hello,

first of all i want to thank you for the great integration of the surepet products in Iobroker.
Unfortunately my Iobroker log is daily flooded with approximately 40 - 70 error messages.

Here is an Example:

```

2023-08-04 08:21:11.857 - error: sureflap.0 (14831) Request timeout
2023-08-04 08:21:11.857 - error: sureflap.0 (14831) Error: Request timeout: retrying in 60 seconds
2023-08-04 08:21:11.858 - error: sureflap.0 (14831) Request error: Error: socket hang up
2023-08-04 08:23:12.166 - error: sureflap.0 (14831) Request timeout
2023-08-04 08:23:12.167 - error: sureflap.0 (14831) Error: Request timeout: retrying in 60 seconds
2023-08-04 08:23:12.167 - error: sureflap.0 (14831) Request error: Error: socket hang up
2023-08-04 08:25:13.161 - error: sureflap.0 (14831) Request timeout
2023-08-04 08:25:13.161 - error: sureflap.0 (14831) Error: Request timeout: retrying in 60 seconds
2023-08-04 08:25:13.162 - error: sureflap.0 (14831) Request error: Error: socket hang up
2023-08-04 08:27:14.169 - error: sureflap.0 (14831) Request timeout
2023-08-04 08:27:14.169 - error: sureflap.0 (14831) Error: Request timeout: retrying in 60 seconds
2023-08-04 08:27:14.170 - error: sureflap.0 (14831) Request error: Error: socket hang up
2023-08-04 08:29:14.486 - error: sureflap.0 (14831) Request timeout
2023-08-04 08:29:14.487 - error: sureflap.0 (14831) Error: Request timeout: retrying in 60 seconds
2023-08-04 08:29:14.487 - error: sureflap.0 (14831) Request error: Error: socket hang up
2023-08-04 08:31:14.790 - error: sureflap.0 (14831) Request timeout
2023-08-04 08:31:14.791 - error: sureflap.0 (14831) Error: Request timeout: retrying in 60 seconds
2023-08-04 08:31:14.791 - error: sureflap.0 (14831) Request error: Error: socket hang up
```

The installed version of the Adapter is the latest V 1.1.9
JS-controller         5.0.11
Node.js            v18.17.0
NPM                      9.6.7

This kind of problem consists for a long time but i think over the time it got worse and worse. 
I don't know if i'm the only one facing this problems but i hope there is a way to solve this.

Please let me know if there is a way to thank you for your work with a little donation.

**Comment by Matis-11** [comment]
Same here, 1.1.8 & 1.1.9
JS: 4.0.24
node: 18.17.0
NPM: 9.6.7

`
2023-08-05 07:20:42.458  - error: sureflap.0 (4360) Request timeout
--
2023-08-05 07:20:42.489  - error: sureflap.0 (4360) Error: Request timeout: retrying in 60 seconds
2023-08-05 07:20:42.495  - error: sureflap.0 (4360) Request error: Error: socket hang up
2023-08-05 07:22:44.180  - error: sureflap.0 (4360) Request timeout
2023-08-05 07:22:44.181  - error: sureflap.0 (4360) Error: Request timeout: retrying in 60 seconds
2023-08-05 07:22:44.183  - error: sureflap.0 (4360) Request error: Error: socket hang up
2023-08-05 07:25:31.256  - error: sureflap.0 (4360) Request timeout
2023-08-05 07:25:31.257  - error: sureflap.0 (4360) Error: Request timeout: retrying in 60 seconds
2023-08-05 07:25:31.260  - error: sureflap.0 (4360) Request error: Error: socket hang up
`

**Comment by Sickboy78** [comment]
Hi!

I have currently the same problem, running
JS-controller 4.0.24
Node.js v18.17.0
NPM 9.6.7

But not so frequent:
30.07. - 7
31.07. - 12
01.08. - 7
02.08. - 9
03.08. - 15
04.08. - 33
today so far - 11 errors

Scince I didn't change anything regarding the https requests for month, I can only guess, the surepet API has some problems and/or has become slow.
Only thing I did recently was updating my node.js from version 14.x to 18.x. I hope it doesn't have something todo with the node version ...
If somebody could check, if the App also has problems at the same time, when the errors occur in the log, that would point to problems at the surepet API. We already had this before.
Anyway, I increased the already high timeout of 60s to 90s. (You can test it by directly installing the latest version from url.)
Lets give it some days and see, if this reduces the errors.

Best Regards
Sebastian

@MAR-L3Y Thanks, but I'm getting by. Seeing that my adapter is appreciated is reward enough.
If you want to donate something, give it to your local shelter. They really need it.

**Comment by Matis-11** [comment]
>  I increased the already high timeout of 60s to 90s. 

Much better! Thanks.

**Comment by MAR-L3Y** [comment]
Hi again,

the behavior of the logging is much better now.

But i think like you already said that the API of surepet is quite unstable 

```

sureflap.0 | 2023-08-12 10:17:07.692 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 10:15:03.668 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 10:13:03.236 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 10:10:56.566 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 10:08:11.743 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:54:21.141 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:52:19.066 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:49:53.757 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:47:53.339 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:45:39.903 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:42:51.646 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:40:51.340 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:38:51.019 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:36:50.608 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:34:50.291 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:32:49.874 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:29:29.457 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:27:29.042 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:25:28.625 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:22:58.968 | error | Error: Request returned status code 500
sureflap.0 | 2023-08-12 09:21:28.852 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:19:26.102 | error | Error: Request returned status code 404
sureflap.0 | 2023-08-12 09:18:08.513 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:16:08.090 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:14:07.675 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:12:07.265 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:10:05.664 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:08:05.359 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:06:04.931 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 09:00:12.646 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:57:46.328 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:55:46.019 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:53:43.091 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:50:55.957 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:48:55.645 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:46:30.384 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:44:27.154 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:41:33.980 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:39:25.943 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:37:21.937 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 08:08:47.052 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 07:33:36.341 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 07:30:33.931 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 07:28:32.030 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 07:26:31.609 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 07:24:31.187 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 07:22:30.767 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 07:19:24.330 | error | Error: Request returned status code 504
sureflap.0 | 2023-08-12 07:16:35.606 | error | Error: Request returned status code 504

```
is it possible to push this status code 504 errors to just show in the debug logging of the adapter?
or just to give a error message to the log if there are more than X errors in 1 hour?
for me the adapter works perfectly fine, some times if i look at the surepet app and try to set something there  and i have to do it twice or more times - i don`t wonder why there are so many errors in the connection between api and adapter ; )


Thanks again for your great work - as wished i just donated 15€ to our local shelter.

Have a nice Weekend

Cheers
Marcus

**Comment by Sickboy78** [comment]
Hi!

> i just donated 15€ to our local shelter.

Thats really nice, thank you!

In v1.2.0 I increased the timeout further to 120s.
And I changed the logging of connection error's to debug, if the same error occurs multiple times.
So only the first error will be logged as error and the following as debug, until another error occurs.

I close this issue but I really hope, Surepet is working on the performence of its API.

Best Regards
Sebastian
